diff --git a/SPECS/log4j.spec b/SPECS/log4j.spec
index 9e7c320..7632e18 100644
--- a/SPECS/log4j.spec
+++ b/SPECS/log4j.spec
@@ -1,4 +1,5 @@
 %bcond_without  jp_minimal
+AutoReqProv:    no
 
 Name:           log4j
 Version:        2.17.2
@@ -13,6 +14,33 @@ URL:            https://logging.apache.org/%{name}
 Source0:        %{name}-%{version}.tar.gz
 
 Patch2:         logging-log4j-Remove-unsupported-EventDataConverter.patch
+Requires:       (java-headless or java-11-headless)
+Requires:       javapackages-filesystem
+Provides:       mvn(log4j:log4j) = 2.17.2
+Provides:       mvn(log4j:log4j:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-1.2-api) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-1.2-api:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-api) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-api-java9:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-api-java9:zip:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-api:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-core) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-core-java9:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-core-java9:zip:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-core:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-jpl) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-jpl:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-osgi) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-osgi:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-to-jul) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-to-jul:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j:pom:) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.1.2-api) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.api) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.core) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.jpl) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.osgi) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.to-jul) = 2.17.2
 
 BuildRequires:  maven-local
 BuildRequires:  mvn(com.lmax:disruptor)
@@ -76,18 +104,46 @@ variety of output targets.
 
 %package slf4j
 Summary:        Binding between LOG4J 2 API and SLF4J
+AutoReqProv:    no
+Requires:       (java-headless or java-11-headless)
+Requires:       javapackages-filesystem
+Requires:       mvn(org.apache.logging.log4j:log4j-api) = 2.17.2
+Requires:       mvn(org.apache.logging.log4j:log4j-core) = 2.17.2
+Requires:       mvn(org.slf4j:slf4j-api)
+Provides:       mvn(org.apache.logging.log4j:log4j-slf4j-impl) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-slf4j-impl:pom:) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-to-slf4j) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-to-slf4j:pom:) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.slf4j-impl) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.to-slf4j) = 2.17.2
 
 %description slf4j
 Binding between LOG4J 2 API and SLF4J.
 
 %package jcl
 Summary:        Apache Log4j Commons Logging Bridge
+AutoReqProv:    no
+Requires:       (java-headless or java-11-headless)
+Requires:       javapackages-filesystem
+Requires:       mvn(commons-logging:commons-logging)
+Requires:       mvn(org.apache.logging.log4j:log4j-api) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-jcl) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-jcl:pom:) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.jcl) = 2.17.2
 
 %description jcl
 Apache Log4j Commons Logging Bridge.
 
 %package web
 Summary:        Apache Log4j Web
+AutoReqProv:    no
+Requires:       (java-headless or java-11-headless)
+Requires:       javapackages-filesystem
+Requires:       mvn(org.apache.logging.log4j:log4j-api) = 2.17.2
+Requires:       mvn(org.apache.logging.log4j:log4j-core) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-web) = 2.17.2
+Provides:       mvn(org.apache.logging.log4j:log4j-web:pom:) = 2.17.2
+Provides:       osgi(org.apache.logging.log4j.web) = 2.17.2
 
 %description web
 Support for Log4j in a web servlet container.
